packages/core/src/actions/enqueueActions.ts

```diff
@@ -135,7 +136,16 @@ function resolveEnqueueActions(
   const enqueue: Parameters<typeof collect>[0]['enqueue'] = function enqueue(
     action
   ) {
-    actions.push(action);
+    actions.push(
+      convertAction(
+        action as any,
+        snapshot.machine.root,
+        'enqueue' + Math.random(), // TODO: this should come from state node ID which isn't provided
```

Comment:
how hard would it be to fix it here?

Reply:
👋

Reply:
If you do the refactor where we don't convert actions eagerly, should we try to fix this still?

Reply:
ye, this becomes obsolete when we don't convert actions eagerly